gamedata/alldefs_post.lua

```diff
@@ -648,19 +648,104 @@ function UnitDef_Post(name, uDef)
+	
+	categories["OBJECT"] = function(unitDef) return string.find(unitDef.category, "OBJECT") end
+	categories["ALL"] = function() return true end
+	categories["MOBILE"] = function(unitDef) return unitDef.speed and unitDef.speed > 0 end
+	categories["NOTMOBILE"] = function(unitDef) return not categories.MOBILE(unitDef) end
+	categories["WEAPON"] = function(unitDef) return unitDef.weapondefs ~= nil end
+	categories["NOWEAPON"] = function(unitDef) return not categories.WEAPON(unitDef) end
+	categories["VTOL"] = function(unitDef) return unitDef.canfly == true end
+	categories["NOTAIR"] = function(unitDef) return not categories.VTOL(unitDef) end
+	categories["HOVER"] = function(unitDef) return hoverList[unitDef.movementclass] and unitDef.maxwaterdepth and unitDef.maxwaterdepth < 1 end -- convertible tank/boats have maxwaterdepth
+	categories["NOTHOVER"] = function(unitDef) return not categories.HOVER(unitDef) end
+	categories["SHIP"] = function(unitDef) return shipList[unitDef.movementclass] or (hoverList[unitDef.movementclass] and unitDef.maxwaterdepth and unitDef.maxwaterdepth >=1) end
+	categories["NOTSHIP"] = function(unitDef) return not categories.SHIP(unitDef) end
+	categories["NOTSUB"] = function(unitDef) return not subList[unitDef.movementclass] end
+	categories["CANBEUW"] = function(unitDef) return amphibList[unitDef.movementclass] end
+	categories["UNDERWATER"] = function(unitDef) return (unitDef.minwaterdepth and unitDef.waterline == nil) or (unitDef.minwaterdepth and unitDef.waterline > unitDef.minwaterdepth) end
+	categories["SURFACE"] = function(unitDef) return not categories.UNDERWATER(unitDef) and not categories.VTOL(unitDef) end
+	categories["MINE"] = function(unitDef) return unitDef.weapondefs and unitDef.weapondefs.minerange end
+	categories["COMMANDER"] = function(unitDef) return commanderList[unitDef.movementclass] end
+	categories["EMPABLE"] = function(unitDef) return categories.SURFACE(unitDef) and unitDef.customparams and unitDef.customparams.paralyzemultiplier ~= 0 end
+	
+	
+	for name, unitDef in pairs(UnitDefs) do
+		if string.find(unitDef.category, "OBJECT") then
+		else
+	
+			-- temrorary code, pending unitdef cleanup
+			local isT4AIR
+			local isLIGHTAIRSCOUT
+			local isGROUNDSCOUT
+			local isRAPTOR
+			if manualCategory(unitDef, "T4AIR") then isT4AIR = true end
+			if manualCategory(unitDef, "LIGHTAIRSCOUT") then isLIGHTAIRSCOUT = true end
+			if manualCategory(unitDef, "GROUNDSCOUT") then isGROUNDSCOUT = true end
+			if manualCategory(unitDef, "RAPTOR") then isRAPTOR = true end
+			unitDef.category = ""
+			if isT4AIR == true then append(unitDef, " T4AIR") end
+			if isLIGHTAIRSCOUT == true then append(unitDef, " LIGHTAIRSCOUT") end
+			if isGROUNDSCOUT == true then append(unitDef, " GROUNDSCOUT") end
+			if isRAPTOR == true then append(unitDef, " isRAPTOR") end
+			if name == "armmex" or name == "cormex" or name == "legmex" or name == "legmext15" then append(unitDef, " CANBEUW") end
```

Comment:
put `can_be_UwU` into mex customparams and check for it inside the `categories["CANBEUW"]` function